boards: Added new board Raytac mdbt50q-db-33

RaytacStanley opened the pull request:
Because it is a bit messy to add together with the previous mdbt50q-db-40, so I separate mdbt50q-db-33.

RaytacStanley committed:
boards: Added new board Raytac mdbt50q-db-33

Because it is a bit messy to add together with the previous mdbt50q-db-40, so I separate mdbt50q-db-33.

RaytacStanley committed:
boards: Added new board Raytac mdbt50q-db-33

Because it is a bit messy to add together with the previous mdbt50q-db-40, so I separate mdbt50q-db-33.

nordicjm reviewed (changes requested):
Same comments as in https://github.com/zephyrproject-rtos/zephyr/pull/52657

RaytacStanley committed:
boards: Fix the problem that the image is too large

Fix the problem that the MDOB50Q-DB-33 image is too large. Reduce image size to 50KB.

RaytacStanley committed:
boards: Fix the problem that the image is too large

Fix the problem that the MDOB50Q-DB-33 image is too large. Reduce image size to 50KB.

RaytacStanley committed:
Merge branch 'zephyr-mdbt50q-db-33' of https://github.com/RaytacStanley/zephyr into zephyr-mdbt50q-db-33

RaytacStanley committed:
boards: rebase branch

This branch is out-of-date with the base branch.
After I performed rebase to update this branch, there was a conflict with another branch of mine. So I manually modify the file test_adc.c .
